os/arch/arm/src/armv8-m/up_assert.c

```diff
@@ -119,6 +119,8 @@ extern sq_queue_t g_freemsg_list;
 #ifdef CONFIG_APP_BINARY_SEPARATION
 extern uint32_t g_assertpc;
 #endif
+
+bool recursive_abort = false;
```

Comment:
Do you think we should check for abort_mode flag as well at the start of up_assert().
This will prevent any other recursive abort scenarios as well ? (Similar to armv7-r)

Reply:
The current code has the same effect. Basically recursive_abort needs to be set to true in assert handler.